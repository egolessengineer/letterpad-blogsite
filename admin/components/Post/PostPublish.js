import React, { Component } from "react";
import { notify } from "react-notify-toast";
import PostActions from "./PostActions";
import { gql, graphql } from "react-apollo";
import moment from "moment";
import config from "../../../config";
import { UPDATE_POST_QUERY } from "../../../shared/queries/Mutations";

const actions = {
    publish: "Published",
    draft: "Save Draft"
};
class PostPublish extends Component {
    constructor(props) {
        super(props);
        this.changePostStatus = this.changePostStatus.bind(this);
        this.changeSlug = this.changeSlug.bind(this);
        this.state = {
            post: this.props.post,
            published: 0
        };
    }
    componentWillReceiveProps(nextProps) {
        const status = nextProps.post.status == "publish" ? 1 : 0;
        this.setState({ published: status });
    }

    componentDidMount() {
        PostActions.setData(this.props.post);
    }

    changePostStatus(e) {
        this.setState({
            published: ~~e.target.checked
        });
    }
    changeSlug(e) {
        this.state.post.slug = e.target.value;
        this.setState(this.state);
        PostActions.setData({ slug: this.state.post.slug });
    }

    async updatePost(e, status) {
        e.preventDefault();
        PostActions.setData(status);
        let data = PostActions.getData();
        const update = await this.props.update({
            ...this.props.post,
            ...data
        });
        if (update.data.updatePost.ok) {
            // If any component has subscribed to get notifications on update/delete, it will be notified.
            PostActions.postUpdated(update.data.updatePost.post);
            if (this.props.create) {
                return notify.show("Post created", "success", 3000);
            }
            if (this.props.post.status === "trash") {
                return notify.show("Post trashed", "success", 3000);
            }
            this.setState({ post: update.data.updatePost.post });
            return notify.show("Post updated", "success", 3000);
        }
        let errors = update.data.updatePost.errors;
        if (errors && errors.length > 0) {
            errors = errors.map(error => error.message);
            notify.show(errore.join("\n"), "error", 3000);
        }
    }

    getButton(label, btnType = "btn-dark", status) {
        if (typeof status == "undefined") {
            status = this.state.published ? "publish" : "draft";
        }
        if (status)
            return (
                <div className="btn-item">
                    <button
                        type="submit"
                        onClick={e => this.updatePost(e, { status: status })}
                        className={"btn btn-sm " + btnType}
                    >
                        {label}
                    </button>
                </div>
            );
    }

    render() {
        const publishedCls = this.state.published ? "on" : "off";
        const permalink =
            config.rootUrl + this.state.post.type + "/" + this.state.post.slug;
        const actionLabel = this.props.create ? "Create" : "Update";
        return (
            <div className="card">
                <div className="module-title">Publishing</div>
                <div className={"switch-block m-b-20 " + publishedCls}>
                    <span className="switch-label switch-off-text">Draft</span>
                    <label className="switch">
                        <input
                            type="checkbox"
                            onChange={this.changePostStatus}
                            checked={this.state.published}
                        />
                        <span className="slider round" />
                    </label>
                    <span className="switch-label switch-on-text">Publish</span>
                </div>
                <div className="x_content m-b-20">
                    <label className="custom-label">
                        <i className="fa fa fa-calendar" /> Published at
                    </label>
                    <input
                        type="text"
                        className="form-control"
                        placeholder="Published date"
                        value={moment(
                            new Date(this.state.post.created_at)
                        ).format("DD-MM-Y hh:mm A")}
                    />
                </div>
                <div className="x_content m-b-20">
                    <label className="custom-label">
                        <i className="fa fa fa-link" /> Slug
                    </label>
                    <input
                        type="text"
                        className="form-control"
                        placeholder="Slug"
                        defaultValue={this.state.post.slug}
                        onBlur={this.changeSlug}
                    />
                </div>
                <div className="x_content m-b-20">
                    <label className="custom-label">
                        <i className="fa fa fa-link" /> Permalink
                    </label>
                    <div>
                        <a target="_blank" href={permalink}>
                            {permalink}
                        </a>
                    </div>
                </div>
                <div className="x_content">
                    <div className="btn-together">
                        {this.getButton(actionLabel, "btn-dark")}
                        {this.getButton("Trash", "btn-danger", "trash")}
                    </div>
                </div>
            </div>
        );
    }
}

const updateQueryWithData = graphql(UPDATE_POST_QUERY, {
    props: ({ mutate }) => ({
        update: data =>
            mutate({
                variables: data,
                updateQueries: {
                    getPost: (prev, { mutationResult }) => {
                        return {
                            post: {
                                ...prev.post,
                                ...mutationResult.data.updatePost.post
                            }
                        };
                    }
                }
            })
    })
});
export default updateQueryWithData(PostPublish);
