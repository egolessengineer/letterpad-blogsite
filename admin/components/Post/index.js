export { default as ArticleCreate } from "./ArticleCreate";
export { default as ArticleEdit } from "./ArticleEdit";
export { default as Categories } from "./Categories";
export { default as ContentEditable } from "./ContentEditable";
export { default as Editor } from "./Editor";
export { default as Excerpt } from "./Excerpt";
export { default as FeaturedImage } from "./FeaturedImage";
export { default as PostFilters } from "./Filter";
export { default as PostActions } from "./PostActions";
export { default as PostPublish } from "./PostPublish";
export { default as PostRows } from "./TableRow";
export { default as Tags } from "./Tags";
