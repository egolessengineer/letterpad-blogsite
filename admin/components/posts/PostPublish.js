import React, { Component } from "react";
import { notify } from "react-notify-toast";
import PostActions from "./PostActions";
import { gql, graphql } from "react-apollo";
import moment from "moment";
import siteConfig from "../../../config/site.config";
import { UPDATE_POST_QUERY } from "../../../shared/queries/Mutations";
import TextField from "material-ui/TextField";
import Button from "material-ui/Button";
import Card, { CardHeader, CardContent, CardActions } from "material-ui/Card";
//import DatePicker from "material-ui/DatePicker";
import Switch from "material-ui/Switch";

const actions = {
    publish: "Published",
    draft: "Save Draft"
};
class PostPublish extends Component {
    constructor(props) {
        super(props);
        this.changePostStatus = this.changePostStatus.bind(this);
        this.changeSlug = this.changeSlug.bind(this);
        this.handleDateChange = this.handleDateChange.bind(this);
        this.state = {
            post: this.props.post,
            published: 1
        };
    }
    componentWillReceiveProps(nextProps) {
        const status = nextProps.post.status == "publish" ? 1 : 0;
        this.setState({ published: status });
    }

    componentDidMount() {
        PostActions.setData(this.props.post);
        if (this.props.post.created_at) {
            this.state.post.created_at = moment(
                new Date(this.props.post.created_at)
            ).format("DD-MM-Y hh:mm A");
            this.setState(this.state);
        }
    }

    changePostStatus(e, checked) {
        this.setState({
            published: ~~checked
        });
    }
    changeSlug(e) {
        this.state.post.slug = e.target.value;
        this.setState(this.state);
        PostActions.setData({ slug: this.state.post.slug });
    }
    handleDateChange(e) {
        this.state.post.created_at = e.target.value;
        this.setState(this.state);
        PostActions.setData({ created_at: e.target.value });
    }

    async updatePost(e, status) {
        e.preventDefault();
        PostActions.setData(status);
        let data = PostActions.getData();
        const update = await this.props.update({
            ...this.props.post,
            ...data
        });
        if (update.data.updatePost.ok) {
            // If any component has subscribed to get notifications on update/delete, it will be notified.
            PostActions.postUpdated(update.data.updatePost.post);
            if (this.props.create) {
                return notify.show("Post created", "success", 3000);
            }
            if (this.props.post.status === "trash") {
                return notify.show("Post trashed", "success", 3000);
            }
            update.data.updatePost.post.created_at = moment(
                new Date(update.data.updatePost.post.created_at)
            ).format("DD-MM-Y hh:mm A");

            this.setState({ post: update.data.updatePost.post });
            return notify.show("Post updated", "success", 3000);
        }
        let errors = update.data.updatePost.errors;
        if (errors && errors.length > 0) {
            errors = errors.map(error => error.message);
            notify.show(errore.join("\n"), "error", 3000);
        }
    }

    getButton(label, status, isPrimary) {
        if (typeof status == "undefined") {
            status = this.state.published ? "publish" : "draft";
        }

        return (
            <Button
                onClick={e => this.updatePost(e, { status: status })}
                color={isPrimary ? "primary" : ""}
            >
                {label}
            </Button>
        );
    }

    render() {
        const publishedCls = this.state.published ? "on" : "off";
        const permalink =
            siteConfig.root_url +
            this.state.post.type +
            "/" +
            this.state.post.slug;
        const actionLabel = this.props.create ? "Create" : "Update";
        return (
            <Card>
                <CardHeader title="Publishing" />
                <CardContent>
                    <div className={"switch-block m-b-20 " + publishedCls}>
                        <span className="switch-label switch-off-text">
                            Draft
                        </span>
                        <Switch
                            checked={!!this.state.published}
                            onChange={this.changePostStatus}
                        />
                        <span className="switch-label switch-on-text">
                            Publish
                        </span>
                    </div>
                    <div className="x_content m-b-20">
                        <label className="custom-label">
                            <i className="fa fa fa-calendar" /> Published at
                        </label>
                        <TextField
                            value={this.state.post.created_at}
                            hintText="Published date"
                            onChange={this.handleDateChange}
                            fullWidth={true}
                        />
                    </div>
                    <div className="x_content m-b-20">
                        <label className="custom-label">
                            <i className="fa fa fa-link" /> Slug
                        </label>
                        <TextField
                            value={this.state.post.slug}
                            hintText="Slug"
                            onChange={this.changeSlug}
                            fullWidth={true}
                        />
                    </div>
                    <div className="x_content m-b-20">
                        <label className="custom-label">
                            <i className="fa fa fa-link" /> Permalink
                        </label>
                        <div>
                            <a target="_blank" href={permalink}>
                                {permalink}
                            </a>
                        </div>
                    </div>
                </CardContent>
                <CardActions>
                    {this.getButton(actionLabel, "publish", true)}
                    {this.getButton("Trash", "draft", false)}
                </CardActions>
            </Card>
        );
    }
}

const updateQueryWithData = graphql(UPDATE_POST_QUERY, {
    props: ({ mutate }) => ({
        update: data =>
            mutate({
                variables: data,
                updateQueries: {
                    getPost: (prev, { mutationResult }) => {
                        return {
                            post: {
                                ...prev.post,
                                ...mutationResult.data.updatePost.post
                            }
                        };
                    }
                }
            })
    })
});
export default updateQueryWithData(PostPublish);
