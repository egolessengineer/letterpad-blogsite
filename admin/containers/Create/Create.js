import React, { Component } from "react";
import { graphql } from "react-apollo";
import moment from "moment";
import PropTypes from "prop-types";
import { plural } from "../../../shared/util";
import {
    ArticleCreate,
    PostPublish,
    PostActions,
    Tags,
    Categories,
    Excerpt
} from "../../components/Post";
import { CREATE_POST } from "../../../shared/queries/Mutations";

class Create extends Component {
    constructor(props) {
        super(props);
        this.state = {
            loading: true,
            post: {}
        };
    }
    componentWillMount() {
        const { type } = this.props;
        const title = "Draft - " + moment().format("L LT");
        this.props.createPost({ type, title }).then(result => {
            PostActions.setData(result.data.createPost.post);
            this.setState({
                loading: false,
                post: result.data.createPost.post
            });
        });

        PostActions.subscribe(post => {
            if ((post.status = "trash")) {
                this.props.history.push(`/admin/${plural[post.type]}`);
            } else {
                this.props.history.push(
                    `/admin/${plural[post.type]}/${post.id}`
                );
            }
        });
        document.body.classList.add(`create-${type}`);
    }
    render() {
        if (this.state.loading) {
            return <div>hello</div>;
        }
        return (
            <section className="module-xs">
                <div className="row">
                    <div className="col-lg-8 column">
                        <ArticleCreate post={this.state.post} />
                    </div>
                    <div className="col-lg-4 column">
                        <PostPublish create post={this.state.post} />
                        <Tags post={this.state.post} />
                        <Categories post={this.state.post} />
                        <Excerpt post={this.state.post} />
                    </div>
                </div>
            </section>
        );
    }
}

Create.propTypes = {
    history: PropTypes.object,
    createPost: PropTypes.func,
    type: PropTypes.string
};

const createQueryWithData = graphql(CREATE_POST, {
    props: ({ mutate }) => ({
        createPost: data =>
            mutate({
                variables: data
            })
    })
});

export default createQueryWithData(Create);
