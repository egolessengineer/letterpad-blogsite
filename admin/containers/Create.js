import React, { Component } from "react";
import { graphql } from "react-apollo";
import ArticleCreate from "../components/posts/ArticleCreate";
import PostPublish from "../components/posts/PostPublish";
import PostActions from "../components/posts/PostActions";
import Tags from "../components/posts/Tags";
import Categories from "../components/posts/Categories";
import { CREATE_POST } from "../../shared/queries/Mutations";
import Excerpt from "../components/posts/Excerpt";
import { plural } from "../../shared/util";
import Loader from "../components/Loader";

class Create extends Component {
    constructor(props) {
        super(props);
        this.state = {
            loading: true,
            post: {}
        };
    }
    componentDidMount() {
        const { type } = this.props;
        this.props.createPost({ type }).then(result => {
            PostActions.setData(result.data.createPost.post);
            this.setState({
                loading: false,
                post: result.data.createPost.post
            });
        });

        PostActions.subscribe(post => {
            if ((post.status = "trash")) {
                this.props.history.push(`/admin/${plural[post.type]}`);
            } else {
                this.props.history.push(
                    `/admin/${plural[post.type]}/${post.id}`
                );
            }
        });
        document.body.classList.add(`create-${type}`);
    }
    render() {
        if (this.state.loading) {
            return <Loader />;
        }

        return (
            <div>
                <div className="col-lg-8 column">
                    <ArticleCreate post={this.state.post} />
                </div>
                <div className="col-lg-4 column">
                    <PostPublish create post={this.state.post} />
                    <br />
                    <Tags post={this.state.post} />
                    <br />
                    <Categories post={this.state.post} />
                    <br />
                    <Excerpt post={this.state.post} />
                </div>
            </div>
        );
    }
}

const createQueryWithData = graphql(CREATE_POST, {
    props: ({ mutate }) => ({
        createPost: data =>
            mutate({
                variables: data
            })
    })
});

export default createQueryWithData(Create);
