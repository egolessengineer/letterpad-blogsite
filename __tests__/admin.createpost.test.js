import React from "react";
import { Create } from "../admin/containers/Create/Create";
import { PostPublish } from "../admin/components/Post/PostPublish";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { PostActions } from "../admin/components/Post";
Enzyme.configure({ adapter: new Adapter() });

describe("Creating post", () => {
    test("Test if post gets created with new data", done => {
        const createPost = props => {
            return Promise.resolve({
                data: {
                    createPost: {
                        post: {
                            title: "draft title",
                            body: "blog post",
                            excerpt: "excerpt",
                            cover_image: "",
                            author_id: 1,
                            type: props.type,
                            status: "draft",
                            slug: "hello-world",
                            mode: "rich-text",
                            mdPreview: ""
                        }
                    }
                }
            });
        };
        const wrapper = shallow(<Create createPost={createPost} type="post" />);
        process.nextTick(() => {
            wrapper.update();
            expect(wrapper).toMatchSnapshot();
            done();
        });
    });

    test("Test if post gets created and redirected to Edit", done => {
        const post = {
            title: "draft title",
            body: "blog post",
            excerpt: "excerpt",
            cover_image: "",
            author_id: 1,
            type: "post",
            status: "draft",
            slug: "hello-world",
            mode: "rich-text",
            mdPreview: ""
        };

        const wrapper = shallow(
            <PostPublish
                post={post}
                create
                history={{
                    push: () => {
                        done(); // this should run for the test to pass.
                    }
                }}
                toggleActionDrawer={() => {}}
                update={data => {
                    return Promise.resolve({
                        data: {
                            updatePost: {
                                ok: true,
                                post: data
                            }
                        }
                    });
                }}
            />
        );

        wrapper
            .find(".publish-btn")
            .first()
            .simulate("click");
    });
});
