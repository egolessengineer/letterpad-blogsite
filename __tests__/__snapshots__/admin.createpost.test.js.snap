// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creating post Test if post gets created with new data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Create
    createPost={[Function]}
    type="post"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row"
        >
          <Link
            className="toggle-zenview"
            onClick={[Function]}
            replace={false}
            to="#"
          >
            <i
              className="fa fa-eye"
            />
          </Link>
          <div
            className="col-lg-12 distractor"
          >
            <Apollo(PostPublish)
              create={true}
              history={undefined}
              post={
                Object {
                  "author_id": 1,
                  "body": "blog post",
                  "cover_image": "",
                  "excerpt": "excerpt",
                  "mdPreview": "",
                  "mode": "rich-text",
                  "slug": "hello-world",
                  "status": "draft",
                  "title": "draft title",
                  "type": "post",
                }
              }
              toggleActionDrawer={[Function]}
            />
          </div>
          <div
            className="col-lg-6 column article-holder"
          >
            <CreateArticle
              post={
                Object {
                  "author_id": 1,
                  "body": "blog post",
                  "cover_image": "",
                  "excerpt": "excerpt",
                  "mdPreview": "",
                  "mode": "rich-text",
                  "slug": "hello-world",
                  "status": "draft",
                  "title": "draft title",
                  "type": "post",
                }
              }
            />
          </div>
          <div
            className="col-lg-6 column preview distractor"
          >
            <MdPreview
              post={
                Object {
                  "author_id": 1,
                  "body": "blog post",
                  "cover_image": "",
                  "excerpt": "excerpt",
                  "mdPreview": "",
                  "mode": "rich-text",
                  "slug": "hello-world",
                  "status": "draft",
                  "title": "draft title",
                  "type": "post",
                }
              }
              preview=""
            />
          </div>
        </div>,
        <PostActionDrawer
          isOpen={false}
          post={
            Object {
              "author_id": 1,
              "body": "blog post",
              "cover_image": "",
              "excerpt": "excerpt",
              "mdPreview": "",
              "mode": "rich-text",
              "slug": "hello-world",
              "status": "draft",
              "title": "draft title",
              "type": "post",
            }
          }
          toggleActionDrawer={[Function]}
          toggleFileExplorerModal={[Function]}
        />,
        undefined,
        false,
      ],
      "className": "module-xs create-post",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="toggle-zenview"
              onClick={[Function]}
              replace={false}
              to="#"
            >
              <i
                className="fa fa-eye"
              />
            </Link>,
            <div
              className="col-lg-12 distractor"
            >
              <Apollo(PostPublish)
                create={true}
                history={undefined}
                post={
                  Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  }
                }
                toggleActionDrawer={[Function]}
              />
            </div>,
            <div
              className="col-lg-6 column article-holder"
            >
              <CreateArticle
                post={
                  Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  }
                }
              />
            </div>,
            <div
              className="col-lg-6 column preview distractor"
            >
              <MdPreview
                post={
                  Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  }
                }
                preview=""
              />
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <i
                className="fa fa-eye"
              />,
              "className": "toggle-zenview",
              "onClick": [Function],
              "replace": false,
              "to": "#",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-eye",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Apollo(PostPublish)
                create={true}
                history={undefined}
                post={
                  Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  }
                }
                toggleActionDrawer={[Function]}
              />,
              "className": "col-lg-12 distractor",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "create": true,
                "history": undefined,
                "post": Object {
                  "author_id": 1,
                  "body": "blog post",
                  "cover_image": "",
                  "excerpt": "excerpt",
                  "mdPreview": "",
                  "mode": "rich-text",
                  "slug": "hello-world",
                  "status": "draft",
                  "title": "draft title",
                  "type": "post",
                },
                "toggleActionDrawer": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <CreateArticle
                post={
                  Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  }
                }
              />,
              "className": "col-lg-6 column article-holder",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "post": Object {
                  "author_id": 1,
                  "body": "blog post",
                  "cover_image": "",
                  "excerpt": "excerpt",
                  "mdPreview": "",
                  "mode": "rich-text",
                  "slug": "hello-world",
                  "status": "draft",
                  "title": "draft title",
                  "type": "post",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <MdPreview
                post={
                  Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  }
                }
                preview=""
              />,
              "className": "col-lg-6 column preview distractor",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "post": Object {
                  "author_id": 1,
                  "body": "blog post",
                  "cover_image": "",
                  "excerpt": "excerpt",
                  "mdPreview": "",
                  "mode": "rich-text",
                  "slug": "hello-world",
                  "status": "draft",
                  "title": "draft title",
                  "type": "post",
                },
                "preview": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isOpen": false,
          "post": Object {
            "author_id": 1,
            "body": "blog post",
            "cover_image": "",
            "excerpt": "excerpt",
            "mdPreview": "",
            "mode": "rich-text",
            "slug": "hello-world",
            "status": "draft",
            "title": "draft title",
            "type": "post",
          },
          "toggleActionDrawer": [Function],
          "toggleFileExplorerModal": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
      false,
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
          >
            <Link
              className="toggle-zenview"
              onClick={[Function]}
              replace={false}
              to="#"
            >
              <i
                className="fa fa-eye"
              />
            </Link>
            <div
              className="col-lg-12 distractor"
            >
              <Apollo(PostPublish)
                create={true}
                history={undefined}
                post={
                  Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  }
                }
                toggleActionDrawer={[Function]}
              />
            </div>
            <div
              className="col-lg-6 column article-holder"
            >
              <CreateArticle
                post={
                  Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  }
                }
              />
            </div>
            <div
              className="col-lg-6 column preview distractor"
            >
              <MdPreview
                post={
                  Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  }
                }
                preview=""
              />
            </div>
          </div>,
          <PostActionDrawer
            isOpen={false}
            post={
              Object {
                "author_id": 1,
                "body": "blog post",
                "cover_image": "",
                "excerpt": "excerpt",
                "mdPreview": "",
                "mode": "rich-text",
                "slug": "hello-world",
                "status": "draft",
                "title": "draft title",
                "type": "post",
              }
            }
            toggleActionDrawer={[Function]}
            toggleFileExplorerModal={[Function]}
          />,
          undefined,
          false,
        ],
        "className": "module-xs create-post",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="toggle-zenview"
                onClick={[Function]}
                replace={false}
                to="#"
              >
                <i
                  className="fa fa-eye"
                />
              </Link>,
              <div
                className="col-lg-12 distractor"
              >
                <Apollo(PostPublish)
                  create={true}
                  history={undefined}
                  post={
                    Object {
                      "author_id": 1,
                      "body": "blog post",
                      "cover_image": "",
                      "excerpt": "excerpt",
                      "mdPreview": "",
                      "mode": "rich-text",
                      "slug": "hello-world",
                      "status": "draft",
                      "title": "draft title",
                      "type": "post",
                    }
                  }
                  toggleActionDrawer={[Function]}
                />
              </div>,
              <div
                className="col-lg-6 column article-holder"
              >
                <CreateArticle
                  post={
                    Object {
                      "author_id": 1,
                      "body": "blog post",
                      "cover_image": "",
                      "excerpt": "excerpt",
                      "mdPreview": "",
                      "mode": "rich-text",
                      "slug": "hello-world",
                      "status": "draft",
                      "title": "draft title",
                      "type": "post",
                    }
                  }
                />
              </div>,
              <div
                className="col-lg-6 column preview distractor"
              >
                <MdPreview
                  post={
                    Object {
                      "author_id": 1,
                      "body": "blog post",
                      "cover_image": "",
                      "excerpt": "excerpt",
                      "mdPreview": "",
                      "mode": "rich-text",
                      "slug": "hello-world",
                      "status": "draft",
                      "title": "draft title",
                      "type": "post",
                    }
                  }
                  preview=""
                />
              </div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <i
                  className="fa fa-eye"
                />,
                "className": "toggle-zenview",
                "onClick": [Function],
                "replace": false,
                "to": "#",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-eye",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Apollo(PostPublish)
                  create={true}
                  history={undefined}
                  post={
                    Object {
                      "author_id": 1,
                      "body": "blog post",
                      "cover_image": "",
                      "excerpt": "excerpt",
                      "mdPreview": "",
                      "mode": "rich-text",
                      "slug": "hello-world",
                      "status": "draft",
                      "title": "draft title",
                      "type": "post",
                    }
                  }
                  toggleActionDrawer={[Function]}
                />,
                "className": "col-lg-12 distractor",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "create": true,
                  "history": undefined,
                  "post": Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  },
                  "toggleActionDrawer": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CreateArticle
                  post={
                    Object {
                      "author_id": 1,
                      "body": "blog post",
                      "cover_image": "",
                      "excerpt": "excerpt",
                      "mdPreview": "",
                      "mode": "rich-text",
                      "slug": "hello-world",
                      "status": "draft",
                      "title": "draft title",
                      "type": "post",
                    }
                  }
                />,
                "className": "col-lg-6 column article-holder",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "post": Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <MdPreview
                  post={
                    Object {
                      "author_id": 1,
                      "body": "blog post",
                      "cover_image": "",
                      "excerpt": "excerpt",
                      "mdPreview": "",
                      "mode": "rich-text",
                      "slug": "hello-world",
                      "status": "draft",
                      "title": "draft title",
                      "type": "post",
                    }
                  }
                  preview=""
                />,
                "className": "col-lg-6 column preview distractor",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "post": Object {
                    "author_id": 1,
                    "body": "blog post",
                    "cover_image": "",
                    "excerpt": "excerpt",
                    "mdPreview": "",
                    "mode": "rich-text",
                    "slug": "hello-world",
                    "status": "draft",
                    "title": "draft title",
                    "type": "post",
                  },
                  "preview": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isOpen": false,
            "post": Object {
              "author_id": 1,
              "body": "blog post",
              "cover_image": "",
              "excerpt": "excerpt",
              "mdPreview": "",
              "mode": "rich-text",
              "slug": "hello-world",
              "status": "draft",
              "title": "draft title",
              "type": "post",
            },
            "toggleActionDrawer": [Function],
            "toggleFileExplorerModal": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        false,
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
